flax/experimental/nnx/nnx/module.py

```diff
@@ -482,29 +482,10 @@ def sow(
       reduced_value = reduce_fn(init_fn(), value)
       setattr(self, name, variable_type(reduced_value))
 
-  def for_each(
```

Comment:
Should this be called `submodules` and not `module`? Or better yet `iter_submodules`?

I would also recommend adding a few tests.

Reply:
Called it [modules()](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.modules) as it would be familiar to Pytorch users. Wondering if we should try to follow their conventions when possible?